### DISCUSSION TITLE
意図せずレイヤー2に固定されてしまい、レイヤー0に戻らないことがある

### DISCUSSION THREAD

**Discussion by megmilk** [Yowkees/keyball#408] [Q&A] [answered]
お世話になっております
Keyball39を図のようなキーマップを設定しております

![keymap_cheatsheet_keyball39](https://github.com/Yowkees/keyball/assets/54742534/2091cfd0-245c-4c89-ba64-35fc5bf91dbe)

マウスのスクロールを使うためレイヤー移動を次のように行っております

レイヤー0 → レイヤー2 → レイヤー3

この操作を行っていると、何のキーにも触れていない状態であっても、現在のレイヤーがレイヤー2に固定されてしまうことがあります
Keyball本体のLCDを見ると、やはりLayer 2を示しています

私はレイヤー0に基本的なキーを設定しています
なので、どうにうも操作ができなくなり、USBケーブルを抜き差しするしか復旧方法がありません

レイヤー2上にKeyballの再起動ボタンを配置し、この現象が発生した時に押してみました。しかし、いつまで待ってもKeyballは再起動できず
そのためUSBケーブルを抜き差しすることになっています

私のキーマップ設定になにか間違いがありますでしょうか？

ご面倒をおかけしますがご教示いただきたく思います

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

レイヤー切り替えキーについては、いくつか種類があり下記サイト等で説明されています。
https://jayliu50.github.io/qmk-cheatsheet/

TT(2) ですとタップでレイヤーがON/OFFされ、レイヤーが固定されてしまうようですので、MO(2)を使用してみるといかがでしょうか？

**Reply by megmilk** [reply]
@Yowkees ありがとうございます！ 

- レイヤー0の TT(2) を MO(2) に変更
- レイヤー2の TT(3) を MO(3) に変更

このキー設定にしてみました
しばらく使って様子を見てみます

**Reply by megmilk** [reply]
@Yowkees ありがとうございます

下記の設定で解決できました！

> - レイヤー0の TT(2) を MO(2) に変更
> - レイヤー2の TT(3) を MO(3) に変更
